### PR TITLE
[gcp]: Preserve gcp field data in flattened fields

### LINKED ISSUE #2857: [gcp] GKE audit log data differs between GCP integrations
My colleague and I discovered, while building a lab for developing k8 detections, that the GCP generic pub-sub integration provides the full scope of k8 audit data, required to detect malicious or suspicious behavior, while the GCP audit integration only provides a small subset of that data. The GCP generic pub-sub integration supplies the needed data but it does not map the data to fields in ECS leaving it all condensed together in the message field essentially making it useless in regards to writing any detection logic or analysis. The GCP audit integration on the other hand, like I said, supplies only a small subset of the data provided by the GKE audit logs leaving out many critical data fields required for observability and security. Below I have included events from both integrations as an example. You can see that the GCP generic pub-sub k8 audit log event message field contains a great deal more data than the GCP audit log event fields provide. 

[GCP generic pub-sub](https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-module-gcp.html) k8 audit log event:
[gcp-gen-pub-sub-event.txt](https://github.com/elastic/integrations/files/8319125/gcp-gen-pub-sub-event.txt)

vs.

[GCP audit](https://docs.elastic.co/en/integrations/gcp#audit) log event:
[gcp-audit-event.txt](https://github.com/elastic/integrations/files/8319130/gcp-audit-event.txt)

### PR DESCRIPTION
**Pull request by r00tu53r**
## What does this PR do?

* Drop document if it is not an AuditLog type
* Updates the pipeline and saves `protoPayload.request` and `protoPayload.response` in flattened fields
* Re-arrange pipeline processors and group them by on AuditLog fields
* Comment pipeline code with links to GCP object types
* Add data to ECS fields
* Add handling for LogEntryOperation (set event.category and event.type) for long running operations
* Save operation.id to gcp.audit.logentry_operation.id only if it is not the same as .insertId
* Set event.category and event.type to network,configuration =>
   allowed|denied based on authorization_info.resource[].granted

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

* Run pipeline and system tests -
  - `elastic-package test pipeline -v -g`
  - `elastic-package test system -v -g` 

## Related issues

Closes #2857

### CODE REVIEW

File: packages/gcp/data_stream/audit/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -23,92 +60,121 @@ processors:
...
-  - set:
-      field: user.email
-      value: "{{gcp.audit.authentication_info.principal_email}}"
-      if: ctx?.gcp?.audit?.authentication_info?.principal_email != null
   - rename:
       field: json.protoPayload.authenticationInfo.authoritySelector
       target_field: gcp.audit.authentication_info.authority_selector
       ignore_missing: true
+
+# TODO remove - duplicated in client.user.email and client.user.id
```
Comment (efd6) [inline packages/gcp/data_stream/audit/elasticsearch/ingest_pipeline/default.yml:130]: Maybe file an issue for this?

Also, I don't see these fields in the new golden files.

File: packages/gcp/data_stream/audit/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -23,92 +60,121 @@ processors:
...
-            target_field: gcp.audit.request_metadata.raw.caller_ip
-  - rename:
-      field: json.protoPayload.requestMetadata.callerIp
-      target_field: gcp.audit.request_metadata.caller_ip
-      ignore_missing: true
-  - set:
-      field: source.ip
-      value: "{{gcp.audit.request_metadata.caller_ip}}"
-      if: ctx?.gcp?.audit?.request_metadata?.caller_ip != null
+  # TODO remove - duplicated in useragent
```
Comment (efd6) [inline packages/gcp/data_stream/audit/elasticsearch/ingest_pipeline/default.yml:177]: Aggregate with issue above? and the ones below.

File: packages/gcp/data_stream/audit/fields/fields.yml [hunk 3]
```diff
@@ -14,6 +14,9 @@
         - name: authority_selector
           type: keyword
           description: "The authority selector specified by the requestor, if any. It is not guaranteed  that the principal was allowed to use this authority."
+        - name: principal_subject
```
Comment (efd6) [inline packages/gcp/data_stream/audit/fields/fields.yml:17]: I couldn't find this being used anywhere.

File: packages/gcp/data_stream/audit/fields/fields.yml [hunk 4]
```diff
@@ -40,6 +46,23 @@
               type: keyword
               description: |
                 The type of the resource.
+    - name: labels
+      type: flattened
```
Comment (efd6) [inline packages/gcp/data_stream/audit/fields/fields.yml:50]: Provide a description explaining the kinds of data that this will hold?

**Review by efd6 — COMMENTED** [review]
Need to run `elastic-packge build`.

File: packages/gcp/data_stream/audit/elasticsearch/ingest_pipeline/default.yml (hunk 1 above)
Comment (r00tu53r) [reply packages/gcp/data_stream/audit/elasticsearch/ingest_pipeline/default.yml:130]: Fixed. Issue for duplicated field https://github.com/elastic/integrations/issues/3397

File: packages/gcp/data_stream/audit/fields/fields.yml (hunk 3 above)
Comment (r00tu53r) [reply packages/gcp/data_stream/audit/fields/fields.yml:17]: `principalSubject` is available in one test. https://github.com/elastic/integrations/pull/3390/files#diff-5c04d67d3f7d8c1c089e6ae179bc564e77589e390a431b87857b96ef2dfe6e8cL887-R1108

**Review by efd6 — APPROVED** [review]

**Pull request merged**